### PR TITLE
chore: remove all @experimental annotation 

### PR DESCRIPTION
**Pull request by NetaNir**
As a part of v2 efforts we are stabilizing all `@experimental` APIs from stable modules.  A separate PR https://github.com/aws/aws-cdk/pull/14070, adds a lint rule that ban the usage of @experimental.

#### Note on experimental modules
The removal of the @experimental annotation from APIs in experimental modules does not change their stability, as it determined only by their containing module stability.  

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/apigatewayv2.ts
Comment (NetaNir) [inline packages/@aws-cdk/aws-apigateway/lib/apigatewayv2.ts:871]: @nija-at any idea what's this about? Why is this method in a non `.generated` file?

Comment (nija-at) [reply packages/@aws-cdk/aws-apigateway/lib/apigatewayv2.ts:871]: see commit description - https://github.com/aws/aws-cdk/commit/d58667e473c1e07ac8f2f073b3a541569a014e01

**Review by nija-at — APPROVED** [review]
added do-not-merge, to give time for any objections in their modules.

**Comment by skinny85** [comment]
If you could wait for #13780 and #14009 before merging this one, I would appreciate it!

File: packages/aws-cdk/lib/api/aws-auth/credentials.ts
Comment (eladb) [inline packages/aws-cdk/lib/api/aws-auth/credentials.ts:9]: @rix0rrr @NetaNir which APIs exactly are we exporting from the CDK CLI? Are we really ready to call these stable? I am concerned that this will signal to folks that they can use the CLI as a library and will bound is to this APIs... I'd hold off with removing the @experimental tags here (the CLI is also not part of the aws-cdk-lib, so I don't think this is required for v2)

File: packages/@aws-cdk/core/lib/construct-compat.ts
Comment (eladb) [inline packages/@aws-cdk/core/lib/construct-compat.ts:525]: Since some files do not exist in v2-main (like this one), this will result in some merge conflicts. @NetaNir can you please make sure to do a forward merge to v2-main immediately after this is merged so that conflicts will be limited to this change and not intermingled with lots of future changes? (copy @nija-at)

**Review by eladb — APPROVED** [review]
LGTM, please revert the CLI changes. Added 'do-not-merge' to allow others to review.

File: packages/aws-cdk/lib/api/aws-auth/credentials.ts
Comment (NetaNir) [reply packages/aws-cdk/lib/api/aws-auth/credentials.ts:9]: @eladb But it is a part of the AWS CDK, and I don't think there will be any notable distinction in our v2 announcements between the CLI and the framework, at least not in a way that will be easy for customers understand. If we break a public API in the CLI post GA, it will most likely be perceived as not keeping our stability promise. In all honesty, I don't think this annotation was ever understood/noticed by customers.

Comment (nija-at) [reply packages/aws-cdk/lib/api/aws-auth/credentials.ts:9]: @eladb - we should just remove all exports from the package [index](https://github.com/aws/aws-cdk/blob/master/packages/aws-cdk/lib/index.ts) so that this package cannot be used as a library.
(need to confirm if npm/node allows this)

That feels like the correct solution here, no?

**Pull request merged**
